src/client/main.cpp

```diff
@@ -52,6 +52,28 @@ class ProtocachedClient {
     return result;
   }
 
+  ResponseResult Set(const ::std::string &key, const ::std::string &value) {
+    ::protocached::SetRequest request;
+    request.set_key(key);
+    CachedValue cached_value;
+    cached_value.set_value(value);
+    request.mutable_value() -> CopyFrom(cached_value);
```

Comment:
I wonder if copyFrom() can be called on regular CachedValue objects. Then I can use this on the Get method.

Reply:
Yeah, any CachedValue instance would have CopyFrom method. You could actually look at the generated code to see CachedValue implementation.